Modify doc contributing

edithturn opened the pull request:
- Added reference links to docker and docker-compose installation
- To execute ./breeze it is necessary to be inside the airflow folder, specify it in the documentation
- Fixed indentation of number one, and changed from normal text to code the command ./breeze

These fixes make it easier to understand the section "Step 2: Configure Your Environment"

I will appreciate your feedback, this is my first PR.

Commit:
adding reference link for docker and docker-compose installation

Commit:
adding reference link for docker and docker-compose installation

Commit:
adding reference link for docker and docker-compose installation

Commit:
adding reference link for docker and docker-compose installation

Commit:
adding reference link for docker and docker-compose installation

uranusjr commented on CONTRIBUTING.rst:
```diff
@@ -221,10 +221,14 @@ or on macOS with `Homebrew <https://formulae.brew.sh/formula/jq>`_
   brew install jq
 
-3. Enter Breeze: ``./breeze``
+3. Enter Breeze, in airflow folder run:
```
Unintuitively, "airflow folder" actually produces more ambigiouty because there are two such directories, one for the Airflow *source code* and one for the *configuration files* (at `/root/airflow`).

```suggestion
3. Enter Breeze, and run the following in the Airflow source code directory:
```

edithturn replied:
Totally agree with this!

uranusjr commented on CONTRIBUTING.rst:
```diff
@@ -221,10 +221,14 @@ or on macOS with `Homebrew <https://formulae.brew.sh/formula/jq>`_
+
+  ./breeze
+
+Breeze starts with downloading the Airflow CI image from
+the Docker Hub and installing all required dependencies.
```
You need to match indentation here for restructuredText to render correctly.

edithturn replied:
@uranusjr I fixed it, let me know if they are okay. This seems fine :)

uranusjr commented on CONTRIBUTING.rst:
```diff
@@ -221,10 +221,14 @@ or on macOS with `Homebrew <https://formulae.brew.sh/formula/jq>`_
-   the Docker Hub and installing all required dependencies.
+.. code-block:: bash
+
+  ./breeze
```
```suggestion

  .. code-block:: bash

      ./breeze
```

edithturn replied:
I fixed this too!

uranusjr commented on CONTRIBUTING.rst:
```diff
@@ -205,10 +205,10 @@ The Docker env is here to maintain a consistent and common development environme
+and `Docker Compose <https://github.com/apache/airflow/blob/main/BREEZE.rst#docker-compose>`_ and add them to the PATH.
 
-2. Install jq on your machine. The exact command depends on the operating system (or Linux distribution) you use.
+2. Install `jq <https://stedolan.github.io/jq/download/>`_ on your machine. The exact command depends on the operating system (or Linux distribution) you use.
```
Indentation also needs to be fixed here.

Also, consider moving the references out-of-line (external) instead of embedding.

https://docutils.sourceforge.io/docs/user/rst/quickref.html#external-hyperlink-targets

edithturn replied:
I added external links instead of embebed links

edithturn committed:
Update CONTRIBUTING.rst

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

Commit:
indentation and external link added

Commit:
Merge branch 'modify-doc-contributing' of https://github.com/edithturn/airflow into modify-doc-contributing

Commit:
fixing links

Commit:
fixing links

edithturn committed:
Update CONTRIBUTING.rst

Adding external link for docker and docker community
Fixed the indentation.
Clarification of a text: "This will enter the Docker Docker environment "

uranusjr commented on CONTRIBUTING.rst:
```diff
@@ -198,50 +198,63 @@ Step 2: Configure Your Environment
 
 You can use either a local virtual env or a Docker-based env. The differences
-between the two are explained `here <https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#development-environments>`_.
+between the two are explained `here`_. 
```
Let's revert this back to being inline (and use the double-underscore form instead of single).

edithturn replied:
I revert it, thanks for let me know

uranusjr commented on CONTRIBUTING.rst:
```diff
@@ -198,50 +198,63 @@ Step 2: Configure Your Environment
 
-2. Install jq on your machine. The exact command depends on the operating system (or Linux distribution) you use.
-For example, on Ubuntu:
+2. Install `jq` on your machine. The exact command depends on the operating system (or Linux distribution) you use.
```
Can be add a reference to jq as well?

```suggestion
2. Install jq_ on your machine. The exact command depends on the operating system (or Linux distribution) you use.
```

And reference to https://stedolan.github.io/jq/

edithturn replied:
@uranusjr I did it, thak you!

edithturn committed:
Update CONTRIBUTING.rst

I revert the link here, I didn't realize about it. Thank you.
I added the link for jq

edithturn committed:
Update CONTRIBUTING.rst

Completing the url

uranusjr commented on CONTRIBUTING.rst:
```diff
@@ -198,9 +198,7 @@ Step 2: Configure Your Environment
-between the two are explained `here`_. 
-
-.. _here: https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#development-environments/
+between the two are explained `here <https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#development-environments/>`__
```
```suggestion
between the two are explained `here <https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#development-environments/>`__.
```

uranusjr committed:
Fix missing period

edithturn committed:
Update CONTRIBUTING.rst

remvoing space before LOCAL_VIRTUALENV.rst file.

edithturn committed:
Update CONTRIBUTING.rst

Removing the last whitespace

edithturn committed:
removing whitespace with pre-commit
